src/events.js

```diff
@@ -29,9 +29,10 @@ export function humanizeEvents(events) {
 
 function extractEvent(event) {
   return {
-    ...(event.contractId() != null && {
-      contractId: StrKey.encodeContract(event.contractId())
-    }),
+    ...(event.contractId != null &&
```

Comment:
ah, pure js, interface checking by presence of method, may be able to consolidate this to a single conditional, seen a slight variation to look for strictly function presence:
```
typeof event.contractId === 'function'
```

Reply:
Ooh nice call!! I wasn't happy with the double-check but this is a great solution

Reply:
Ah wait no, we still need to check whether or not the contract ID, if the function is present, is actually valid (because `contractId()` might return `null` :facepalm:), since in that case we don't want to include it in the response